src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -19,14 +22,51 @@ class PyTestResult(TestResult):
   pass
 
 
+# TODO: Support deps
+# TODO: Support resources
 @rule(PyTestResult, [Select(HydratedTarget)])
 def run_python_test(target):
-  # TODO: Actually run tests (https://github.com/pantsbuild/pants/issues/6003)
-
-  if 'fail' in target.address.reference():
-    noun = 'failed'
-    status = Status.FAILURE
-  else:
-    noun = 'passed'
-    status = Status.SUCCESS
-  return PyTestResult(status=status, stdout=str('I am a python test which {}'.format(noun)))
+
+  dirs = set()
+  for f in target.adaptor.sources.files_relative_to_buildroot:
+    dir = os.path.dirname(f)
+    if dir == '':
+      dir = '.'
+    dirs.add(dir)
+  dirs = tuple(sorted(dirs))
+
+  # Note that this hasn't been published yet, but needs https://github.com/pantsbuild/pex/pull/605 to be released
+  pex_snapshot = yield Get(Snapshot, UrlToFetch, UrlToFetch("https://github.com/pantsbuild/pex/releases/download/v1.5.2/pex27"))
```

Comment:
TODO: URL should be parameterised

Reply:
John was kind enough to publish `v1.5.2` this past weekend.